spark/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala

```diff
@@ -240,48 +277,90 @@ case class DeltaSharingSource(
       isStartingVersion: Boolean,
       currentLatestVersion: Long): Unit = {
     lastQueryTableTimestamp = System.currentTimeMillis()
+    minUrlExpirationTimestamp = -1
     if (isStartingVersion) {
       // If isStartingVersion is true, it means to fetch the snapshot at the fromVersion, which may
       // include table changes from previous versions.
       val tableFiles = deltaLog.client.getFiles(
         deltaLog.table, Nil, None, Some(fromVersion), None, None
       )
       latestRefreshFunc = () => {
-        deltaLog.client.getFiles(
+        val files = deltaLog.client.getFiles(
           deltaLog.table, Nil, None, Some(fromVersion), None, None
-        ).files.map { f =>
+        ).files
+        var minUrlExpiration: Long = -1L
+        val idToUrl = files.map { f =>
+          if (f.expirationTimestamp != null) {
+            if (minUrlExpiration == -1L) {
+              minUrlExpiration = f.expirationTimestamp
+            } else {
+              minUrlExpiration.min(f.expirationTimestamp)
+            }
+          }
           f.id -> f.url
         }.toMap
+
+        refreshSortedFetchedFiles(idToUrl)
```

Comment:
Does this fix the double fetch issue?

Reply:
yes, it will fix it. This is what you/Ryan suggested.
The refresh only happens CachedTableManager, and will update sortedFetchedFiles